modules/programs/irssi.nix

```diff
@@ -0,0 +1,213 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.irssi;
+
+  ifToYesNo = b: if b then "yes" else "no";
+
+  assignFormat =
+    set:
+      concatStringsSep "\n"
+        (
+          mapAttrsToList (k: v: "  ${k} = \"${v}\";") set
+        );
+
+  chatnetString =
+    concatStringsSep "\n"
+      ( mapAttrsToList
+        (k: v: ''
+          ${k} = {
+            type = "${v.type}";
+            nick = "${v.nick}";
+            autosendcmd = "${concatStringsSep ";" v.autoCmd}";
+          };
+        '')
+        cfg.chatnets
+      );
+
+  serversString =
+    concatStringsSep ",\n"
+      ( mapAttrsToList
+        (k: v: ''
+        {
+          chatnet = "${k}";
+          address = "${v.server.address}";
+          port = "${toString v.server.port}";
+          use_ssl = "${ifToYesNo v.server.useSSL}";
+          ssl_verify = "${ifToYesNo v.server.verifySSL}";
+          autoconnect = "${ifToYesNo v.server.autoconnect}";
+        }'')
+        cfg.chatnets
+      );
+
+  channelString =
+    concatStringsSep ",\n"
+      ( mapAttrsToList
+        (k: v: concatStringsSep ",\n"
+          ( mapAttrsToList
+            (c: cv: ''
+            {
+              chatnet = "${k}";
+              name = "${c}";
+              autojoin = "${ifToYesNo cv.autojoin}";
+            }'')
+            v.channels)
+        )
+        cfg.chatnets
+      );
+
+  serverType =
+    types.submodule (
+      {name, ...}: {
+        options = {
+          address = mkOption {
+            type = types.str;
+            description = "Address of the Chat server.";
+          };
+          port = mkOption {
+            type = types.int;
+            default = 6667;
+            description = "Port of the Chat server.";
+          };
+          useSSL = mkOption {
+            type = types.bool;
+            default = true;
+            description = "Whether SSL should be used.";
+          };
+          verifySSL = mkOption {
+            type = types.bool;
+            default = true;
+            description = "Whether the SSL Certificate should be verified.";
+          };
+          autoconnect = mkOption {
+            type = types.bool;
+            default = false;
+            description = "Whether irssi connects to the server on launch.";
+          };
+        };
+      }
+    );
+
+  channelType =
+    types.submodule (
+      {name, ...}: {
+        options = {
+          name = mkOption {
+            type = types.str;
+            visible = false;
+            default = name;
+            description = "Name of the channel.";
+          };
+          autojoin = mkOption {
+            type = types.bool;
+            default = false;
+            description = "Whether to join this channel on connect.";
+          };
+        };
+      }
+    );
+
+in
+
+{
+
+  options = {
+    programs.irssi = {
+      enable = mkOption {
```

Comment:
```
enable = mkEnableOption "the Irssi chat client";
```

Note, I changed "irssi" to "Irssi" since the https://irssi.org/ website seems to prefer capital i in the beginning. Can change this in the other option descriptions as well.

Reply:
I changed this and switched to `mkEnableOption`.